src/main/scala/mesosphere/marathon/core/storage/store/impl/zk/ZkPersistenceStore.scala

```diff
@@ -351,4 +351,42 @@ class ZkPersistenceStore(
       }
     }
   }
+
+  @SuppressWarnings(Array("all")) // async/await
+  override def startMigration(): Future[Done] = {
+    require(isOpen, "the store must be opened before it can be used")
+
+    async {
+      await(client.create("/migration-in-progress").asTry) match {
+        case Success(_) =>
+          Done
+        case Failure(e: KeeperException.NodeExistsException) =>
+          throw new StoreCommandFailedException(
+            "Migration is already in progress; /migration-in-progress node already exists", e)
```

Comment:
Hm, is the exception really called `NodeExistsException`? This is odd if the node already exists.

Reply:
Why is it odd?

Reply:
This is sensible to me. If you try to create a node, but it already exists, then this is an exception.

This keeps it atomic. If, by chance, two marathon instances were trying to run migrations simultaneously, only one of them would succeed, and the other would fail.

Reply:
Oh jeez. You know, I read `NoNodeExistsException`. Shame on me.